Sources/NIOSOCKS/Channel Handlers/SOCKSClientHandler.swift

```diff
@@ -31,11 +31,16 @@ public final class SOCKSClientHandler: ChannelDuplexHandler {
     private var inboundBuffer: ByteBuffer?
     
     private var bufferedWrites: MarkedCircularBuffer<(NIOAny, EventLoopPromise<Void>?)> = .init(initialCapacity: 8)
+    private let establishHandler: (Channel) -> EventLoopFuture<Void>
```

Comment:
Major nit: I'd probably rename to `establishedHandler` or `connectionEstablishedHandler`. It's a little more clear IMHO.